python/lsst/ts/monochromator/monochromator_csc.py

```diff
@@ -0,0 +1,473 @@
+    FAULT = SALPY_ATMonochromator.ATMonochromator_shared_DetailedState_FaultState
+    MONO_SETTING_UP = SALPY_ATMonochromator.ATMonochromator_shared_DetailedState_MonoSettingUpState
+    STOPPED = SALPY_ATMonochromator.ATMonochromator_shared_DetailedState_StoppedState
+    STOPPED_COOLER_OFF = SALPY_ATMonochromator.ATMonochromator_shared_DetailedState_StoppedCoolerOffState
+    STOPPED_LIGHTS_OFF = SALPY_ATMonochromator.ATMonochromator_shared_DetailedState_StoppedLightOffState
+    STOPPED_LIGHTS_ON = SALPY_ATMonochromator.ATMonochromator_shared_DetailedState_StoppedLightOnState
+    MONO_MANUAL_SETUP = SALPY_ATMonochromator.ATMonochromator_shared_DetailedState_MonoManualSetup
+    MONO_AUTO_SETUP = SALPY_ATMonochromator.ATMonochromator_shared_DetailedState_MonoAutomaticSetup
+
+
+class CSC(salobj.BaseCsc):
+    """
+    Commandable SAL Component (CSC) for the Monochromator.
+    """
+
+    def __init__(self):
+        """
+        Initialize CSC.
+        """
+
+        self._detailed_state = None
+
+        super().__init__(SALPY_ATMonochromator)
+
+        self.detailed_state = self.summary_state
+
+        self.model = Model(self.log)
+
+        # Publish setting versions
+        self.evt_settingVersions.set_put(recommendedSettingsVersion=self.model.recommended_settings,
+                                         recommendedSettingsLabels=self.model.settings_labels)
+
+        self.want_connection = False
+        self._health_loop = None
+
+        self.mock_ctrl = None
+        self.controller_status = None
+
+    def report_detailed_state(self):
+        """Report a new value for detailed_state, including current state.
+        """
+        self.evt_detailedState.set_put(detailedState=self.detailed_state)
+
+    @property
+    def detailed_state(self):
+        """Set or get the detailed state as a `DetailedState` enum.
+
+        If you set the state then it is reported as a detailedState event.
+        You can set detailed_state to a `State` constant or to
+        the integer equivalent.
+
+        Raises
+        ------
+        ValueError
+            If the new summary state is an invalid integer.
+        """
+        return self._detailed_state
+
+    @detailed_state.setter
+    def detailed_state(self, detailed_state):
+        # cast summary_state from an int or State to a State,
+        # and reject invalid int values with ValueError
+        self._detailed_state = DetailedState(detailed_state)
+        self.report_detailed_state()
+
+    def begin_start(self, id_data):
+        """Begin do_start; called before state changes.
+
+        This method call setup on the model, passing the selected setting.
+
+        Parameters
+        ----------
+        id_data : `CommandIdData`
+            Command ID and data
+        """
+        self.model.setup(id_data.data.settingsToApply)
+        self.want_connection = True
+
+    async def do_enable(self, id_data):
```

Comment:
It'd be good to describe what is happening with the method here. Given that you are overriding salobj defaults.